undefined reference to symbol 'pthread_create@@GLIBC_2.2.5' on Ubuntu 18.04

```
root@b2ba6faaf862:/aartfaac2ms/build# cmake .. -DBoost_PYTHON36_LIBRARY=/usr/lib/x86_64-linux-gnu/libboost_python3.so 
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found AOFlagger version 2.15: /usr/lib/x86_64-linux-gnu/libaoflagger.so
-- Found LibXml2: /usr/lib/x86_64-linux-gnu/libxml2.so (found version "2.9.4") 
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.11") 
-- Found PNG: /usr/lib/x86_64-linux-gnu/libpng.so (found version "1.6.34") 
-- Found CFITSIO: /usr/lib/x86_64-linux-gnu/libcfitsio.so  
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Checking for module 'gtkmm-3.0>=3.0.0'
--   Found gtkmm-3.0, version 3.22.2
-- Found all of AOFlagger's dependencies.
--   AOFlagger include dir: /usr/include;/usr/include
--   AOFlagger lib: /usr/lib/x86_64-linux-gnu/libaoflagger.so;/usr/lib/x86_64-linux-gnu/libpng.so;/usr/lib/x86_64-linux-gnu/libz.so;/usr/lib/x86_64-linux-gnu/libfftw3.so;/usr/lib/x86_64-linux-gnu/libcfitsio.so;gtkmm-3.0;atkmm-1.6;gdkmm-3.0;giomm-2.4;pangomm-1.4;glibmm-2.4;gtk-3;gdk-3;pangocairo-1.0;pango-1.0;atk-1.0;cairo-gobject;gio-2.0;cairomm-1.0;cairo;sigc-2.0;gdk_pixbuf-2.0;gobject-2.0;glib-2.0
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython3.6m.so (found suitable version "3.6.9", minimum required is "3") 
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.6.9", minimum required is "3") 
-- Using python version 3.6.9
CMake Warning at /usr/share/cmake-3.10/Modules/FindBoost.cmake:1626 (message):
  No header defined for python36; skipping header check
Call Stack (most recent call first):
  CMakeLists.txt:24 (find_package)


-- Boost version: 1.65.1
-- Found the following Boost libraries:
--   date_time
--   filesystem
--   python36
--   system
CMake Warning at /usr/share/cmake-3.10/Modules/FindBoost.cmake:1626 (message):
  No header defined for numpy36; skipping header check
Call Stack (most recent call first):
  CMakeLists.txt:27 (find_package)


-- Could NOT find Boost
-- Looking for sgemm_
-- Looking for sgemm_ - found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- A library with BLAS API found.
-- Looking for cheev_
-- Looking for cheev_ - not found
-- A library with LAPACK API not found. Please specify library location.
-- Could NOT find HDF5 (missing: HDF5_LIBRARIES HDF5_INCLUDE_DIRS) (found version "")
-- Found the following Casacore components: 
--   ms: /usr/lib/x86_64-linux-gnu/libcasa_ms.so
--   fits: /usr/lib/x86_64-linux-gnu/libcasa_fits.so
--   scimath: /usr/lib/x86_64-linux-gnu/libcasa_scimath.so
--   measures: /usr/lib/x86_64-linux-gnu/libcasa_measures.so
--   scimath_f: /usr/lib/x86_64-linux-gnu/libcasa_scimath_f.so
--   tables: /usr/lib/x86_64-linux-gnu/libcasa_tables.so
--   casa: /usr/lib/x86_64-linux-gnu/libcasa_casa.so
-- Flags passed to C++ compiler: 
-- Configuring done
-- Generating done
-- Build files have been written to: /aartfaac2ms/build
root@b2ba6faaf862:/aartfaac2ms/build# make
Scanning dependencies of target afedit
[  7%] Building CXX object CMakeFiles/afedit.dir/afedit.cpp.o
[ 15%] Linking CXX executable afedit
[ 15%] Built target afedit
Scanning dependencies of target aartfaac2ms
[ 23%] Building CXX object CMakeFiles/aartfaac2ms.dir/main.cpp.o
[ 30%] Building CXX object CMakeFiles/aartfaac2ms.dir/aartfaac2ms.cpp.o
[ 38%] Building CXX object CMakeFiles/aartfaac2ms.dir/aartfaacms.cpp.o
[ 46%] Building CXX object CMakeFiles/aartfaac2ms.dir/averagingwriter.cpp.o
[ 53%] Building CXX object CMakeFiles/aartfaac2ms.dir/fitsuser.cpp.o
[ 61%] Building CXX object CMakeFiles/aartfaac2ms.dir/fitswriter.cpp.o
[ 69%] Building CXX object CMakeFiles/aartfaac2ms.dir/mswriter.cpp.o
[ 76%] Building CXX object CMakeFiles/aartfaac2ms.dir/progressbar.cpp.o
[ 84%] Building CXX object CMakeFiles/aartfaac2ms.dir/stopwatch.cpp.o
[ 92%] Building CXX object CMakeFiles/aartfaac2ms.dir/threadedwriter.cpp.o
[100%] Linking CXX executable aartfaac2ms
/usr/bin/ld: CMakeFiles/aartfaac2ms.dir/aartfaac2ms.cpp.o: undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
//lib/x86_64-linux-gnu/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
CMakeFiles/aartfaac2ms.dir/build.make:363: recipe for target 'aartfaac2ms' failed
make[2]: *** [aartfaac2ms] Error 1
CMakeFiles/Makefile2:104: recipe for target 'CMakeFiles/aartfaac2ms.dir/all' failed
make[1]: *** [CMakeFiles/aartfaac2ms.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,9 +17,12 @@ set(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/CMake)
 find_package(AOFlagger 2 REQUIRED)
 include_directories(${AOFLAGGER_INCLUDE_DIR})
 
+find_package(Threads REQUIRED)
+
 find_package(PythonLibs 3 REQUIRED)
 find_package(PythonInterp 3 REQUIRED)
 message(STATUS "Using python version ${PYTHON_VERSION_STRING}")
+
 set(BOOSTPY ${PYTHON_VERSION_MAJOR}${PYTHON_VERSION_MINOR})
 find_package(Boost 1.55.0 REQUIRED COMPONENTS date_time filesystem python${BOOSTPY})
 set(BOOST_PART_LIBRARIES ${Boost_LIBRARIES})
@@ -27,7 +30,7 @@ set(BOOST_PART_LIBRARIES ${Boost_LIBRARIES})
 find_package(Boost 1.55.0 COMPONENTS numpy${BOOSTPY})
 set(Boost_LIBRARIES ${BOOST_PART_LIBRARIES} ${Boost_LIBRARIES})
 
-SET(CASACORE_MAKE_REQUIRED_EXTERNALS_OPTIONAL TRUE)
+set(CASACORE_MAKE_REQUIRED_EXTERNALS_OPTIONAL TRUE)
 find_package(Casacore REQUIRED COMPONENTS casa ms tables measures fits scimath scimath_f)
 
 include_directories(${CASACORE_INCLUDE_DIRS})
@@ -36,19 +39,19 @@ include_directories(${CASACORE_INCLUDE_DIRS})
 # LD_LIBRARY_PATH doesn't have to be set.
 
 # use, i.e. don't skip the full RPATH for the build tree
-SET(CMAKE_SKIP_BUILD_RPATH  FALSE)
+set(CMAKE_SKIP_BUILD_RPATH  FALSE)
 # when building, don't use the install RPATH already
 # (but later on when installing)
-SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE) 
-SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE) 
+set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
 # add the automatically determined parts of the RPATH
 # which point to directories outside the build tree to the install RPATH
-SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 # the RPATH to be used when installing, but only if it's not a system directory
-LIST(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_PREFIX}/lib" isSystemDir)
-IF("${isSystemDir}" STREQUAL "-1")
-  SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
-ENDIF("${isSystemDir}" STREQUAL "-1")
+list(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_PREFIX}/lib" isSystemDir)
+if("${isSystemDir}" STREQUAL "-1")
+  set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+endif("${isSystemDir}" STREQUAL "-1")
 
 configure_file(version.h.in version.h)
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
@@ -57,10 +60,11 @@ add_executable(aartfaac2ms main.cpp aartfaac2ms.cpp aartfaacms.cpp averagingwrit
 target_link_libraries(aartfaac2ms
 	${AOFLAGGER_LIB} ${CASACORE_LIBRARIES}
 	${Boost_SYSTEM_LIBRARY} ${Boost_DATE_TIME_LIBRARY} ${Boost_PYTHON_LIBRARIES}
-	${PYTHON_LIBRARIES})
+	${PYTHON_LIBRARIES}
+	Threads::Threads)
 
 add_executable(afedit afedit.cpp)
-target_link_libraries(afedit ${CASACORE_LIBRARIES})
+target_link_libraries(afedit ${CASACORE_LIBRARIES} Threads::Threads)
 
 install(TARGETS aartfaac2ms afedit DESTINATION bin)
 

